### ISSUE TITLE
Attempting to bind values to reserved name should trigger a warning

### ISSUE THREAD

**Issue by ogoffart** [slint-ui/slint#739]
### Discussed in https://github.com/sixtyfpsui/sixtyfps/discussions/738

<div type='discussions-op-text'>

<sup>Originally posted by **hbina** December 11, 2021</sup>
I have wasted a disgusting amount of time only to realize that `row` is a reserved keyword. I basically had something like this compiled without warning, which does not even make any sense because `row` here is shadowed by the reserved name anyways.

```60
for row in bson.inner : HorizontalBox {
    key := Text {
        text: row.key;
    }
    value := Text {
        text: row.value;
    }
}
```</div>

**Comment by ogoffart** [comment]
This is actually valid, and we use that quite a lot in our own code:

```
for x in foobar : Text { text: x; }
// or
for item in foobar : SomeStuff { item:item; }
```

The problem in the original example is that key and value element are not direct child.

Eventually we should change the lookup rules: https://github.com/sixtyfpsui/sixtyfps/issues/273

**Comment by ogoffart** [comment]
Lookup changes are done with the new syntax.
Closing this issue now.
